### PR TITLE
Smart contracts whitelist

### PR DESCRIPTION
**Pull request by augustocollado**
### Description

This PR adds a guide to whitelisting smart contracts deployment using polkadot js

### Checklist

- [x] I have added a label to this PR 🏷️
- [ ] I have run my changes through Grammarly
- [ ] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: Smart EVM - Whitelist Smart Contracts Deployment
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:2]:
```suggestion
title: Smart EVM - Whitelisted Smart Contracts Deployment
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+description: Learn how to use Sudo to whitelist smart contracts deployment for your Smart EVM Tanssi appchain, increasing overall security.
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:3]:
```suggestion
description: Learn how to use Sudo to whitelist smart contracts deployers for your Smart EVM Tanssi appchain, increasing overall security.
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+- **Enhanced Security** - by restricting deployment to trusted accounts, the risk of deploying malicious or vulnerable smart contracts is reduced
+- **Quality Assurance** - known and vetted accounts can be required to follow specific coding standards and undergo thorough testing before deployment
+- **Regulatory Compliance** - uses cases that are highly regulated can limit deployment to ensure that smart contracts meet legal and compliance requirements
+- **Spam and Abuse Prevention** - prevent bad actors from deploying large numbers of unnecessary or harmful contracts
+
+In this guide, you'll learn how to use the Sudo account to manage the whitelisted accounts that will be able to deploy smart contracts on your appchain.
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+For the examples in this guide, you will need to have the following:
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:23]: These prerequisites are not in a snippet right?

File: builders/manage/developer-portal/.pages [hunk 4]
```diff
@@ -8,4 +8,5 @@ nav:
   - 'Pause Transactions': 'pause-transactions.md'
   - 'Impersonate Accounts': 'impersonate.md'
   - 'Mint Tokens': 'minting.md'
+  - 'Smart EVM - Whitelist Smart Contracts Deployment': 'smart-contracts-creation-filter.md'
```
Comment (albertov19) [inline builders/manage/developer-portal/.pages:11]:
```suggestion
  - 'Whitelist Smart Contracts Deployers': 'smart-contracts-creation-filter.md'
```

Comment (albertov19) [reply builders/manage/developer-portal/.pages:11]: After rendering the page it does not look that bad. @eshaben what do you think?

image

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+6. If you need to insert more than one account, click on **Add item**
+7. Press **Submit Sudo** and confirm the transaction in your wallet
+
+![Whitelisting Accounts](/images/builders/manage/developer-portal/smart-contracts-creation-filter/smart-contracts-creation-filter-2.webp)
+
+These same steps can be repeated at any moment to remove an account from the whitelist or to add new ones.
+
+## Reestablishing Permission to Deploy Smart Contracts {: #reestablishing-permission}
+
+If you previously authorized some accounts to deploy smart contracts and want to restore the permissions for any account (with enough balance to pay the fees) to deploy smart contracts, then [get your Polkadot.js Apps started](#getting-started) and take the following steps:
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:59]:
```suggestion
If you previously authorized some accounts to deploy smart contracts and want to allow any account to deploy smart contracts (as long as they can cover regular transaction fees), then [get your Polkadot.js Apps started](#getting-started) and take the following steps:
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+## Introduction {: #introduction }
+
+EVM-compatible Tanssi appchains benefit from a unique feature: the appchain governor can define which accounts are authorized to deploy smart contracts, forbidding the action for any other non-whitelisted account.
+
+This feature brings several key benefits that might be a great fit for different use cases or contexts. Some of those benefits are:
+
+- **Enhanced Security** - by restricting deployment to trusted accounts, the risk of deploying malicious or vulnerable smart contracts is reduced
+- **Quality Assurance** - known and vetted accounts can be required to follow specific coding standards and undergo thorough testing before deployment
+- **Regulatory Compliance** - uses cases that are highly regulated can limit deployment to ensure that smart contracts meet legal and compliance requirements
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:16]:
```suggestion
- **Regulatory Compliance** - uses cases that are highly regulated and can limit deployment to ensure that smart contracts meet legal and compliance requirements
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+- **Spam and Abuse Prevention** - prevent bad actors from deploying large numbers of unnecessary or harmful contracts
+
+In this guide, you'll learn how to use the Sudo account to manage the whitelisted accounts that will be able to deploy smart contracts on your appchain.
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:19]:
```suggestion
In this guide, you'll learn how to use the Sudo account to manage the whitelisted accounts that can deploy smart contracts on your appchain.
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+!!! note
+    If you do not see **Sudo** in this menu, then you have not associated the Sudo account with Polkadot.js Apps. Make sure that your [Sudo account is injected by your wallet and connected to Polkadot.js Apps](/builders/manage/developer-portal/sudo/#configuring-polkadotjs-apps){target=\_blank}
+
+## WhiteListing Accounts {: #whitelist-accounts }
+
+To define the accounts that will have authorization to deploy smart contracts, [get your Polkadot.js Apps started](#getting-started) and then take the following steps:
+
+1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the KeyValue parameter
+2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to whitelist the accounts for smart contracts deployment and the later to whitelist the accounts for indirect  (via a smart contract call) smart contract deployment
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:46]:
```suggestion
2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to whitelist the accounts for smart contracts deployment and the latter to whitelist the accounts for indirect  (via a smart contract call) smart contract deployment
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+![Whitelisting Accounts](/images/builders/manage/developer-portal/smart-contracts-creation-filter/smart-contracts-creation-filter-2.webp)
+
+These same steps can be repeated at any moment to remove an account from the whitelist or to add new ones.
+
+## Reestablishing Permission to Deploy Smart Contracts {: #reestablishing-permission}
+
+If you previously authorized some accounts to deploy smart contracts and want to restore the permissions for any account (with enough balance to pay the fees) to deploy smart contracts, then [get your Polkadot.js Apps started](#getting-started) and take the following steps:
+
+1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the KeyValue parameter
+2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to clear the whitelist for smart contracts deployment and the later to clear the whitelist for indirect  (via a smart contract call) smart contract deployment
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:62]:
```suggestion
2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to clear the whitelist for smart contracts deployment and the latter to clear the whitelist for indirect  (via a smart contract call) smart contract deployment
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+## Query the Configuration {: #query-configuration }
+
+To get the current configuration containing the whitelisted accounts that can deploy smart contracts, go to Polkadot.js Apps (as explained in the [Getting Started](#getting-started) section), navigate to the **Developer** Tab, click on **Chain state**, and take the following steps:
+
+1. Select the **parameters** storage
+2. Select the **parameters(ContainerChainTemplateFrontierRuntimeParametersKey)** option
+3. Make sure that the **Include option** switch is on
+4. Make sure that the **ContractDeployFilter** option is selected
+5. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to query the whitelist for smart contracts deployment and the later to query the whitelist for indirect (via a smart contract call) smart contract deployment
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:77]:
```suggestion
5. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to query the whitelist for smart contracts deployment and the latter to query the whitelist for indirect (via a smart contract call) smart contract deployment
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+4. Select the **Whitelisted** option
+5. Insert the whitelisted account
+6. If you need to insert more than one account, click on **Add item**
+7. Press **Submit Sudo** and confirm the transaction in your wallet
+
+![Whitelisting Accounts](/images/builders/manage/developer-portal/smart-contracts-creation-filter/smart-contracts-creation-filter-2.webp)
+
+These same steps can be repeated at any moment to remove an account from the whitelist or to add new ones.
+
+## Reestablishing Permission to Deploy Smart Contracts {: #reestablishing-permission}
```
Comment (albertov19) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:57]:
```suggestion
## Restoring Permissions to Deploy Smart Contracts {: #restoring-permission}
```

**Review by albertov19 — CHANGES_REQUESTED** [review]
Great job!

Minor grammar corrections.

File: builders/manage/developer-portal/smart-contracts-creation-filter.md (hunk 1 above)
Comment (augustocollado) [reply builders/manage/developer-portal/smart-contracts-creation-filter.md:2]: "Whitelist" is intended as a verb here (the other titles in the section start with verbs like manage, dispatch, enable, etc)

Comment (albertov19) [reply builders/manage/developer-portal/smart-contracts-creation-filter.md:2]: Then I would say "Whitelist Smart Contract Deployments"

File: builders/manage/developer-portal/smart-contracts-creation-filter.md (hunk 3 above)
Comment (augustocollado) [reply builders/manage/developer-portal/smart-contracts-creation-filter.md:23]: Not yet, but definitely should.
I'll create a ticket to revisit these guides, create the snippets and update the screenshots

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+Once in Polkadot.js Apps, navigate to the **Developer** Tab and click on **Sudo**.
+
+!!! note
+    If you do not see **Sudo** in this menu, then you have not associated the Sudo account with Polkadot.js Apps. Make sure that your [Sudo account is injected by your wallet and connected to Polkadot.js Apps](/builders/manage/developer-portal/sudo/#configuring-polkadotjs-apps){target=\_blank}
+
+## WhiteListing Accounts {: #whitelist-accounts }
+
+To define the accounts that will have authorization to deploy smart contracts, [get your Polkadot.js Apps started](#getting-started) and then take the following steps:
+
+1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the KeyValue parameter
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:45]:
```suggestion
1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the **keyValue** parameter
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+To follow the next sections of this guide, head to Polkadot.js Apps for your Tanssi appchain. The Polkadot.js Apps link for your Tanssi appchain can be found in your [Tanssi Dashboard](https://apps.tanssi.network){target=\_blank} underneath the **Tooling** section.
+
+![Locating your Polkadot.js Apps Link on apps.tanssi.network](/images/builders/manage/developer-portal/smart-contracts-creation-filter/smart-contracts-creation-filter-1.webp)
+
+Once in Polkadot.js Apps, navigate to the **Developer** Tab and click on **Sudo**.
+
+!!! note
+    If you do not see **Sudo** in this menu, then you have not associated the Sudo account with Polkadot.js Apps. Make sure that your [Sudo account is injected by your wallet and connected to Polkadot.js Apps](/builders/manage/developer-portal/sudo/#configuring-polkadotjs-apps){target=\_blank}
+
+## WhiteListing Accounts {: #whitelist-accounts }
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:41]:
```suggestion
## Whitelisting Accounts {: #whitelist-accounts }
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+To define the accounts that will have authorization to deploy smart contracts, [get your Polkadot.js Apps started](#getting-started) and then take the following steps:
+
+1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the KeyValue parameter
+2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to whitelist the accounts for smart contracts deployment and the latter to whitelist the accounts for indirect  (via a smart contract call) smart contract deployment
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:46]:
```suggestion
2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to whitelist the accounts for smart contract deployments and the latter to whitelist the accounts for indirect  (via a smart contract call) smart contract deployments
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+![Whitelisting Accounts](/images/builders/manage/developer-portal/smart-contracts-creation-filter/smart-contracts-creation-filter-2.webp)
+
+These same steps can be repeated at any moment to remove an account from the whitelist or to add new ones.
+
+## Restoring Permissions to Deploy Smart Contracts {: #restoring-permission}
+
+If you previously authorized some accounts to deploy smart contracts and want to allow any account to deploy smart contracts (as long as they can cover regular transaction fees), then [get your Polkadot.js Apps started](#getting-started) and take the following steps:
+
+1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the KeyValue parameter
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:61]:
```suggestion
1. Select the **parameters** pallet. **setParameter** will be automatically selected in the functions selector and **ContractDeployFilter** in the **keyValue** parameter
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to clear the whitelist for smart contracts deployment and the latter to clear the whitelist for indirect  (via a smart contract call) smart contract deployment
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:62]:
```suggestion
2. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to clear the whitelist for smart contract deployments and the latter to clear the whitelist for indirect  (via a smart contract call) smart contract deployments
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+3. Toggle the **Include option** switch
+4. Select the **All** option
+5. Press **Submit Sudo** and confirm the transaction in your wallet
+
+![Clearing the Whitelisted Accounts](/images/builders/manage/developer-portal/smart-contracts-creation-filter/smart-contracts-creation-filter-3.webp)
+
+## Query the Configuration {: #query-configuration }
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:69]:
```suggestion
## Query the Whitelisted Accounts {: #query-configuration }
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+To get the current configuration containing the whitelisted accounts that can deploy smart contracts, go to Polkadot.js Apps (as explained in the [Getting Started](#getting-started) section), navigate to the **Developer** Tab, click on **Chain state**, and take the following steps:
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:71]:
```suggestion
To get the current configuration containing the whitelisted accounts that can deploy smart contracts, go to Polkadot.js Apps (as explained in the [Getting Started](#getting-started) section), navigate to the **Developer** tab, click on **Chain state**, and take the following steps:
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Select the **parameters** storage
+2. Select the **parameters(ContainerChainTemplateFrontierRuntimeParametersKey)** option
+3. Make sure that the **Include option** switch is on
+4. Make sure that the **ContractDeployFilter** option is selected
+5. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to query the whitelist for smart contracts deployment and the latter to query the whitelist for indirect (via a smart contract call) smart contract deployment
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:77]:
```suggestion
5. Two options will be available in the **ContractDeployFilter** selector: **AllowedAddressesToCreate** and **AllowedAddressesToCreateInner**. Select the **AllowedAddressesToCreate** option if you want to query the whitelist for smart contract deployments and the latter to query the whitelist for indirect (via a smart contract call) smart contract deployments
```

File: builders/manage/developer-portal/.pages [hunk 20]
```diff
@@ -8,4 +8,5 @@ nav:
   - 'Pause Transactions': 'pause-transactions.md'
   - 'Impersonate Accounts': 'impersonate.md'
   - 'Mint Tokens': 'minting.md'
+  - 'Smart EVM - Whitelist Smart Contract Deployments': 'smart-contracts-creation-filter.md'
```
Comment (eshaben) [inline builders/manage/developer-portal/.pages:11]:
```suggestion
  - 'Smart EVM - Whitelist Contract Deployments': 'smart-contracts-creation-filter.md'
```

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 21]
```diff
@@ -0,0 +1,81 @@
+---
+title: Smart EVM - Whitelist Smart Contract Deployments
```
Comment (eshaben) [inline builders/manage/developer-portal/smart-contracts-creation-filter.md:2]:
```suggestion
title: Smart EVM - Whitelist Contract Deployments
```

**Review by eshaben — COMMENTED** [review]
Nice! Just small things 🙂

File: builders/manage/developer-portal/smart-contracts-creation-filter.md [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+description: Learn how to use Sudo to whitelist smart contracts deployers for your Smart EVM Tanssi appchain, increasing overall security.
```
[the remark this answers is not in the document]
Comment (augustocollado) [reply builders/manage/developer-portal/smart-contracts-creation-filter.md]: We don't have a definition from mkt on this I think

**Review by albertov19 — APPROVED** [review]

**Review by eshaben — APPROVED** [review]
LGTM! ✨

**Pull request merged**